SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialViewWorkerConnection.cpp

```diff
@@ -0,0 +1,168 @@
+#include "Interop/Connection/SpatialViewWorkerConnection.h"
+
+#include "SpatialGDKSettings.h"
+#include "SpatialView/CommandRequest.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/ConnectionHandler/SpatialOSConnectionHandler.h"
+#include "SpatialView/ViewCoordinator.h"
+
+namespace
+{
+
+SpatialGDK::ComponentData ToComponentData(FWorkerComponentData* Data)
+{
+	return SpatialGDK::ComponentData(SpatialGDK::OwningComponentDataPtr(Data->schema_type), Data->component_id);
+}
+
+SpatialGDK::ComponentUpdate ToComponentUpdate(FWorkerComponentUpdate* Update)
+{
+	return SpatialGDK::ComponentUpdate(SpatialGDK::OwningComponentUpdatePtr(Update->schema_type), Update->component_id);
+}
+
+}  // anonymous namespace
+
+void USpatialViewWorkerConnection::SetConnection(Worker_Connection* WorkerConnectionIn)
+{
+	TUniquePtr<SpatialGDK::SpatialOSConnectionHandler> Handler = MakeUnique<SpatialGDK::SpatialOSConnectionHandler>(WorkerConnectionIn);
+	Coordinator = MakeUnique<SpatialGDK::ViewCoordinator>(MoveTemp(Handler));
+}
+
+void USpatialViewWorkerConnection::FinishDestroy()
+{
+	Coordinator.Reset();
+	Super::FinishDestroy();
+}
+
+void USpatialViewWorkerConnection::DestroyConnection()
+{
+	Coordinator.Reset();
+}
+
+TArray<SpatialGDK::OpList> USpatialViewWorkerConnection::GetOpList()
+{
+	check(Coordinator.IsValid());
+	Coordinator->FlushMessagesToSend();
```

Comment:
Is there a reason you flush here?

Reply:
Will check. I think this is left over from a debug thing. Nice spot.

Reply:
Actually found this was something I'd meant to come aback to. Cleaned up the flush code generally a little bit; now it should always flush messages to send in TickFlush regardless of what settings you have.

Reply:
Will need to check that the stuff I did was actually correct, although this was a fairly serious mistake 😓